### PR TITLE
Use 'exec' as the default qualifier when registering modules

### PR DESCRIPTION
**Pull request by ericbottard**
Will need to `FLUSHDB` your redis to remove all keys.

### CODE REVIEW

File: spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/config/ArtifactRegistryPopulator.java [hunk 1]
```diff
@@ -115,8 +125,8 @@ private void populateDefault(String name, ArtifactType type) {
 	 * @return default coordinates for the provided module
 	 */
 	private ArtifactCoordinates defaultTaskCoordinatesFor(String moduleName) {
-		return ArtifactCoordinates.parse(String.format("%s:%s:%s",
-				DEFAULT_TASK_GROUP_ID, moduleName, DEFAULT_VERSION));
+		return ArtifactCoordinates.parse(String.format("%s:%s:%s:%s:%s",
+				DEFAULT_TASK_GROUP_ID, moduleName, DEFAULT_EXTENSION, DEFAULT_CLASSIFIER, DEFAULT_VERSION));
```
Comment (mbogoevici) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/config/ArtifactRegistryPopulator.java]: Should we use 'exec' for all kinds of artifacts - or is this affecting only modules ?

Comment (ericbottard) [inline spring-cloud-dataflow-admin/src/main/java/org/springframework/cloud/dataflow/admin/config/ArtifactRegistryPopulator.java]: this is only affecting OOTB `stream` and `task` registrations.

Newly introduced libraries would typically be hand-registered, for now.

**Comment by markfisher** [comment]
with this change can we remove the use of `DEFAULT_MODULE_CLASSIFIER` in the `ModuleLauncher`? (would be good, since currently it's impossible to use a module that does not use "exec") cc @mbogoevici

**Comment by mbogoevici** [comment]
@markfisher we probably should - that assumption belongs at a higher level anyway

**Comment by markfisher** [comment]
Agreed. And as long as we are populating the registry with those classifiers, I think we should remove the assumption at the `ModuleLauncher` level.

**Comment by markfisher** [comment]
LGTM; merging

**Pull request merged**